data/World/Dodongos Cavern MQ.json

```diff
@@ -42,7 +42,61 @@
                 can_play(Song_of_Time) and (can_child_attack or is_adult)",
             "Dodongos Cavern MQ GS Larvae Room": "can_use(Sticks) or has_fire_source",
             "Dodongos Cavern MQ GS Lizalfos Room": "can_blast_or_smash",
-            "Dodongos Cavern MQ Deku Scrub Staircase": "can_stun_deku"
+            "Dodongos Cavern MQ Deku Scrub Staircase": "can_stun_deku",
+            "Dodongos Cavern MQ Torch Puzzle Room Recovery Heart": "can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire)",
```

Comment:
It took my a very long time to find Dodongos Cavern MQ Torch Puzzle Room Recovery Heart. It's invisible! I was under the impression that invisible freestandings were not shuffled? If that's the case then it should probably be removed from the shuffle. (I saw an image where the shuffled freestanding was visible in-game, which would be incorrect behaviour.)

As far as the logic goes, it should match with the compass chest, where the only new thing blocking you is actually a skulltula in the staircase room, so: is_adult or can_child_attack or Nuts.

Reply:
I not sure if this thread should be marked as resolved b/c I think there's still a general discussion about shuffle the normally-invisible freestandings.